agent/src/main/scala/io/scalac/mesmer/agent/akka/persistence/AkkaPersistenceAgent.scala

```diff
@@ -24,67 +21,36 @@ object AkkaPersistenceAgent
    * @return
    *   Resulting agent and resulting configuration based on runtime properties
    */
-  override protected def agent(
-    config: AkkaPersistenceModule.AkkaPersistenceMetricsDef[Boolean],
-    jars: AkkaPersistenceModule.AkkaPersistenceJars[Version]
-  ): (Agent, AkkaPersistenceModule.AkkaPersistenceMetricsDef[Boolean]) = {
+  def agent: Agent = {
 
-    val recoveryTimeAgent         = if (config.recoveryTime) recoveryTime(jars) else None
-    val recoveryTotalAgent        = if (config.recoveryTotal) recoveryTotal(jars) else None
-    val persistentEventAgent      = if (config.persistentEvent) persistentEvent(jars) else None
-    val persistentEventTotalAgent = if (config.persistentEventTotal) persistentEventTotal(jars) else None
-    val snapshotAgent             = if (config.snapshot) snapshot(jars) else None
+    val config = module.enabled
 
-    val resultantAgent =
-      recoveryTimeAgent.getOrElse(Agent.empty) ++
-        recoveryTotalAgent.getOrElse(Agent.empty) ++
-        persistentEventAgent.getOrElse(Agent.empty) ++
-        persistentEventTotalAgent.getOrElse(Agent.empty) ++
-        snapshotAgent.getOrElse(Agent.empty)
-
-    val enabled = AkkaPersistenceModule.Impl(
-      recoveryTime = recoveryTimeAgent.isDefined,
-      recoveryTotal = recoveryTotalAgent.isDefined,
-      persistentEvent = persistentEventAgent.isDefined,
-      persistentEventTotal = persistentEventTotalAgent.isDefined,
-      snapshot = snapshotAgent.isDefined
-    )
-
-    (resultantAgent, enabled)
-  }
-
-  def agent(config: Config): Agent = {
-    def orEmpty(condition: Boolean, agent: Agent): Agent = if (condition) agent else Agent.empty
+    // TODO can we introduce some foldable and/or functor/applicative to be able to map over those without manually checking every element here
```

Comment:
> Do you think it can be done in this PR? Maybe no need to leave a TODO?

I don't think so - I'll just remove TODO and add a ticket for this. We later on can decide if we want to spend time on this.

> We can at least do this: ...

Yeah, sure, this seems to alleviate at least some pains 🙃